core/src/Text/Interpolation/Nyan/Lens.hs

```diff
@@ -0,0 +1,52 @@
+-- SPDX-FileCopyrightText: 2022 Serokell <https://serokell.io/>
+--
+-- SPDX-License-Identifier: MPL-2.0
+
+module Text.Interpolation.Nyan.Lens
+  ( module Text.Interpolation.Nyan.Lens.TH
+  , module Text.Interpolation.Nyan.Lens.Type
+  , (^.)
+
+  , (%~)
+  , (%=)
+
+  , (.~)
+  , (.=)
+
+  , (?~)
+  , (?=)
+
+  , (&~)
+  )
+  where
+
+import Control.Monad.State (MonadState, State, execState, modify)
+import Control.Applicative (Const(..))
+import Data.Functor.Identity (Identity(..))
+
+import Text.Interpolation.Nyan.Lens.TH
+import Text.Interpolation.Nyan.Lens.Type
+
+(^.) :: s -> Getting a s a -> a
+s ^. l = getConst $ l Const s
+
+(%~) :: ASetter s t a b -> (a -> b) -> s -> t
+l %~ f = runIdentity . l (Identity . f)
+
+(%=) :: MonadState s m => ASetter s s a b -> (a -> b) -> m ()
+l %= f = modify (l %~ f)
+
+(.~) :: ASetter s t a b -> b -> s -> t
+l .~ b = runIdentity . l (const $ Identity b)
+
+(.=) :: MonadState s m => ASetter s s a b -> b -> m ()
+l .= b = modify (l .~ b)
+
+(?~) :: ASetter s t a (Maybe b) -> b -> s -> t
+l ?~ b = l .~ (Just b)
+
+(?=) :: MonadState s m => ASetter s s a (Maybe b) -> b -> m ()
+l ?= b = modify (l ?~ b)
+
+(&~) :: s -> State s a -> s
+s &~ l = execState l s
```

Comment:
Ah, I think actually we don't need _operators_ and other helpers, especially if this is a functionality that we will have to cover with tests (we would really have to). We may need `Lens` type, but only if it will be used to define our new lenses.

The point is not to provide a simple replacement for `lens` library, but rather to be compatible with `lens` without depending on it. So that the user who wants to use our new lenses to configure the options, could use them along with `lens` package; and the user who does not want to use lens could go completely without them.